src/Functions/FunctionsOpDate.cpp

```diff
@@ -53,7 +53,6 @@ class FunctionOpDate : public IFunction
     }
 
     bool useDefaultImplementationForConstants() const override { return true; }
-    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0, 2}; }
```

Comment:
Why did you remove 2, but didn't add any test for that?

Reply:
Because `2` is a 3rd argument and we only accept 2